src/dao/users.dao.ts

```diff
@@ -1,39 +1,58 @@
+
+import { Op } from 'sequelize';
 import { User, UserModel } from "../models/User";
 import { RegisterUserDto } from '../dto/register.user.dto';
 import { UserDto, createUserDto } from '../dto/user.dto';
-import  { Op } from 'sequelize';
-import bcryptjs from 'bcryptjs';
+import { Profile } from '../models/Profile';
+import { Company } from '../models/Company';
+import { sequelize } from '../instances/sequalize';
+
 class UsersDao{ 
-     registerUser = async(registerUserDto: RegisterUserDto): Promise<number> => {
-      const user: UserModel =  await User.create(registerUserDto);      
-      return user.id;
+     registerUser = async ( registerUserDto: RegisterUserDto ): Promise<number> => { 
+        const name = registerUserDto.profile.company.name;
+        User.addHook( 'afterCreate', async ( user, options ) => {
+        await Company.update({ name, companyOwner: user.getDataValue( 'id' ) }, {
```

Comment:
Should Company also needs to be created here?

Reply:
I realized I forgot to remove the hook at the end.

![image](https://user-images.githubusercontent.com/43373641/153932726-69d0a6d3-3a67-4d4e-8b01-dbc2eba80ef8.png)

https://github.com/tatjama/uteam-api/blob/a38c8f83108adfc436b53d57bfbac07ce04446e6/src/dao/users.dao.ts#L11-L32